Pre-Processing/get_non_soft_story_images.py

```diff
@@ -39,14 +46,23 @@ def get_image(address):
 
     for row in reader:
         row_number += 1
-        address = row[0]
+        if row_number < start_row:
+            continue
+        if row_number > end_row:
+            break
+        address = row[address_column]
         image = get_image(address)
+        if image is None:
+            continue
         np_image = Image.open(io.BytesIO(image))
         np_image = np.asarray(np_image)
         def on_press(event):
```

Comment:
Define your functions separately from your main logic (which you can put under `if __name__ == "__main__":` - see previous note)